internal/cluster/distribution/eks/eksprovider/workflow/bootstrap_activity.go

```diff
@@ -162,22 +164,31 @@ func (a *BootstrapActivity) Execute(ctx context.Context, input BootstrapActivity
 		mapRoles += fmt.Sprintf(mapRolesTemplate, roleMeta+"/"+roleID)
 	}
 
-	awsAuthConfigMap := v1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{Name: "aws-auth"},
-		Data: map[string]string{
-			"mapRoles": mapRoles,
-			"mapUsers": fmt.Sprintf(mapUsersTemplate, input.ClusterUserArn, userNameFromArn(input.ClusterUserArn)),
-		},
+	var mapUsers = fmt.Sprintf(mapUsersTemplate, input.ClusterUserArn, userNameFromArn(input.ClusterUserArn))
+	var defaultAWSConfigMap = fmt.Sprintf(`apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: aws-auth
+  namespace: kube-system
+data:
+  mapRoles: |
+    %s
+  mapUsers: |
+    %s`, mapRoles, mapUsers)
+
+	mergedConfigMap, err := mergeAuthConfigMaps(defaultAWSConfigMap, input.AuthConfigMap)
+	if err != nil {
+		return nil, errors.WrapIf(err, "failed to merge config map")
 	}
 
-	_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Create(&awsAuthConfigMap)
+	_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Create(mergedConfigMap)
 	if k8serr.ReasonForError(err) == metav1.StatusReasonAlreadyExists {
-		_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Update(&awsAuthConfigMap)
+		_, err = kubeClient.CoreV1().ConfigMaps("kube-system").Update(mergedConfigMap)
 		if err != nil {
-			return nil, errors.WrapIfWithDetails(err, "failed to create config map", "configmap", awsAuthConfigMap.Name)
+			return nil, errors.WrapIfWithDetails(err, "failed to create config map", "configmap", mergedConfigMap.Name)
```

Comment:
**Question**: shouldn't this error message be `failed to **update** config map`?

Reply:
Thanks, fixing it!